google/cloud/storage/tools/run_testbench_utils.sh

```diff
@@ -75,18 +75,28 @@ start_testbench() {
   echo "Launching testbench emulator in the background"
   trap kill_testbench EXIT
 
-  # The tests typically run in a Docker container, where the ports are largely
-  # free; when using in manual tests, you can set EMULATOR_PORT.
-  local -r testbench_port=${TESTBENCH_PORT:-8000}
-
-  gunicorn --bind 0.0.0.0:${testbench_port} \
+  gunicorn --bind 0.0.0.0:0 \
       --worker-class gevent \
       --access-logfile - \
       --pythonpath "${PROJECT_ROOT}/google/cloud/storage/testbench" \
       testbench:application \
       >testbench.log 2>&1 </dev/null &
   TESTBENCH_PID=$!
 
+  local testbench_port=""
+  local -r listening_at='Listening at: http://0.0.0.0:\([1-9][0-9]*\)'
+  for attempt in $(seq 1 8); do
+    testbench_port=$(sed -n "s,^.*${listening_at}.*$,\1,p" testbench.log)
```

Comment:
Opt: The "^.*" and ".*$" don't really contribute anything.

Reply:
If you mean `^` and `$` I agree, but at least they help me figure out what is going on.  The `.*` preceding and after the `${listening_at}` are needed because there is a lot more on that line (some of which changes), e.g.:

```
[2019-02-09 18:39:20 +0000] [9180] [INFO] Listening at: http://0.0.0.0:8000 (9180)
```

Reply:
The `.*`s are only needed because of the `^` and `$`, but OK.